clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -135,8 +137,10 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
     // per-shard counts for error reporting; XXX: we could add our own global
     // counts to compare to the serial stream).
     ++shard->ref_count_;
-    if (type_is_instr(memref.instr.type))
+    if (type_is_instr(memref.instr.type)) {
         ++shard->instr_count_;
+        ++shard->instr_count_since_last_timestamp_;
```

Comment:
(Storing the count as of the last timestamp is more efficient, but an extra increment shouldn't matter so let's leave it.)